cmd/cluster/transferowner.go

```diff
@@ -0,0 +1,430 @@
+	transferOwnerCmd.MarkFlagRequired("old-organization-id")
+
+	return transferOwnerCmd
+}
+
+func newTransferOwnerOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *transferOwnerOptions {
+	return &transferOwnerOptions{
+		IOStreams:     streams,
+		GlobalOptions: globalOpts,
+	}
+}
+
+func (o *transferOwnerOptions) complete(cmd *cobra.Command, _ []string) error {
+	o.output = o.GlobalOptions.Output
+	err := validateOrgID(o.newOrganizationId)
+	if err != nil {
+		return fmt.Errorf("error validating new organization-id")
+	}
+	err = validateOrgID(o.oldOrganizationId)
+	if err != nil {
+		return fmt.Errorf("error validating new organization-id")
+	}
+	return nil
+}
+
+// basic organization-id input validation
+func validateOrgID(orgID string) error {
+	ok, err := regexp.MatchString(`[a-zA-Z0-9]{27}`, orgID)
+	if err != nil {
+		return fmt.Errorf("error validating new organization-id")
+	}
+	if !ok {
+		return fmt.Errorf("invalid new organization id provided: %v", orgID)
+	}
+	return nil
+}
+
+func (o *transferOwnerOptions) run() error {
+	fmt.Print("Before making changes in OCM, the cluster must have the pull secret updated to be the new owner's. ")
+	fmt.Print("See: https://github.com/openshift/ops-sop/blob/master/v4/howto/replace-pull-secret.md\n")
+	utils.ConfirmSend()
+
+	// Create an OCM client to talk to the cluster API
+	// the user has to be logged in (e.g. 'ocm login')
+	ocm := utils.CreateConnection()
+	defer func() {
+		if err := ocm.Close(); err != nil {
+			fmt.Printf("Cannot close the ocm (possible memory leak): %q", err)
+		}
+	}()
+
+	// Find and setup all resources that are needed
+
+	cluster, err := utils.GetCluster(ocm, o.clusterID)
+	if err != nil {
+		return err
+	}
+
+	externalClusterID, ok := cluster.GetExternalID()
+	if !ok {
+		return fmt.Errorf("cluster has no external id")
+	}
+
+	subscription, err := utils.GetSubscription(ocm, externalClusterID)
+	if err != nil {
+		return fmt.Errorf("could not get subscription: %w", err)
+	}
+
+	account, err := utils.GetAccount(ocm, o.newOwnerName)
+	if err != nil {
+		return fmt.Errorf("could not get new owners account: %w", err)
+	}
+
+	accountID, ok := account.GetID()
+	if !ok {
+		return fmt.Errorf("account has no id")
+	}
+
+	oldOwnerAccount, err := utils.GetAccount(ocm, o.oldOwnerName)
+	if err != nil {
+		return fmt.Errorf("could not get old owners account: %w", err)
+	}
+
+	subscriptionID, ok := subscription.GetID()
+	if !ok {
+		return fmt.Errorf("subscription has no id")
+	}
+
+	clusterConsole, ok := cluster.GetConsole()
+	if !ok {
+		return fmt.Errorf("cluster has no console url")
+	}
+
+	clusterURL, ok := clusterConsole.GetURL()
+	if !ok {
+		return fmt.Errorf("cluster has no console url")
+	}
+
+	displayName, ok := subscription.GetDisplayName()
+	if !ok {
+		return fmt.Errorf("subscription has no displayName")
+	}
+
+	ok = validateOldOwner(o, subscription, oldOwnerAccount)
```

Comment:
This is to catch wrong input. It is allowed to fail, so that if for example the subscription got patched but then the  command failed, the rerun of the command does not get blocked when checking against the already updated subscription.